apex/contrib/test/layer_norm/test_fast_rms_norm.py

```diff
@@ -0,0 +1,265 @@
+        self.stop_ = torch.cuda.Event(enable_timing=True)
+        self.stream_ = stream
+
+    def start(self):
+        self.stream_.record_event(self.start_)
+
+    def stop(self):
+        self.stream_.record_event(self.stop_)
+
+    def sync(self):
+        self.stream_.synchronize()
+
+    def millis(self):
+        return self.start_.elapsed_time(self.stop_)
+
+
+def size_in_bytes(t):
+    return torch.numel(t) * t.element_size()
+
+
+def metrics(y_ref, y, epsilon=1e-6):
+    y_ref = y_ref.float()
+    y = y.float()
+    relerr, mse = (
+        (y_ref - y).abs().sum() / (y_ref.abs().sum() + epsilon),
+        (y_ref - y).square().mean(),
+    )
+    return relerr.item(), mse.item()
+
+
+device = torch.device("cuda")
+fp32 = torch.float32
+fp16 = torch.float16
+bf16 = torch.bfloat16
+
+
+def backward_(dz, x, rs, gamma):
+
+    wtype = gamma.dtype
+    itype = x.dtype
+    otype = dz.dtype
+    ctype = rs.dtype
+    rs = rs.unsqueeze(1)
+
+    hidden_size = gamma.numel()
+    y = rs * x.to(ctype)
+    dgamma = (dz * y).view(-1, hidden_size).sum(0, dtype=ctype)
+    dy = dz.view(-1, hidden_size).to(ctype) * gamma.unsqueeze(0).to(ctype)
+
+    mdyy = (dy * y).mean(1, keepdim=True, dtype=ctype)
+    dx = rs * (dy - mdyy * y)
+
+    return dx.to(itype), dgamma.to(wtype)
+
+
+def benchmark_(S, B, hidden_size, itype, wtype, runs=100):
+    epsilon = 1e-5
+
+    x = torch.randn((S * B, hidden_size), dtype=itype, device=device)
+    # beta = torch.randn(hidden_size, dtype=wtype, device=device)
+    gamma = torch.randn(hidden_size, dtype=wtype, device=device)
+    dz = torch.randn(x.shape, dtype=wtype, device=device)
+
+    stream = torch.cuda.Stream()
+    with torch.cuda.stream(stream):
+
+        timer = GPUTimer(stream)
+
+        # warmup
+        for r in range(runs):
+            z, rsigma = fln.rmsnorm_fwd(x, gamma, epsilon)
+
+        timer.start()
+        for r in range(runs):
+            z, rsigma = fln.rmsnorm_fwd(x, gamma, epsilon)
+        timer.stop()
+        timer.sync()
+
+        total_bytes_fwd = sum([size_in_bytes(t) for t in [x, z, gamma, rsigma]])
+
+        ms_fwd = timer.millis() / runs
+
+        print(
+            "[FWD] Time: {:.4f}ms Throughput: {:.4f} GB/sec".format(
+                ms_fwd, total_bytes_fwd * 1e-6 / ms_fwd
+            )
+        )
+
+        timer.start()
+        for r in range(runs):
+            dx, dgamma, dgp = fln.rmsnorm_bwd(dz, x, rsigma, gamma)
+        timer.stop()
+        timer.sync()
+
+        total_bytes_bwd = sum(
+            [
+                size_in_bytes(t)
+                for t in [dz, x, rsigma, gamma, dx, dgamma, dgp, dgp]
+            ]
+        )
+
+        ms_bwd = timer.millis() / runs
+
+        print(
+            "[BWD] Time: {:.4f}ms Throughput: {:.4f} GB/sec".format(
+                ms_bwd, total_bytes_bwd * 1e-6 / ms_bwd
+            )
+        )
+
+def _test_impl(S, B, hidden_size, itype, wtype, ctype=fp32):
```

Comment:
I understand this test is based on the existing one for faster layer norm but could you access the following two items?
1. make this function a member method of the test class below
2. use `torch.testing.assert_close` instead of `check_err` and `TestFastRMSNorm.assertAll`